src/httpserver/handlers/handler.clj

```diff
@@ -0,0 +1,22 @@
+(ns httpserver.handlers.handler
+  (:require [httpserver.handlers.handler_function_selector :as selector]))
+
+(import '(com.td.HttpServer HttpResponse))
+(import '(com.td.HttpServer IHandler))
+(import '(com.td.HttpServer HttpRequest))
+
+(deftype main-handler []
+  com.td.HttpServer.IHandler
+  (generateResponse [this httpRequest]
+    (let [request (hash-map :path (.path httpRequest) :method (.method httpRequest) :headers (.headers httpRequest) :body (.body httpRequest))
```

Comment:
I feel like this is complex enough to be extracted into a private function, so that this can just read:

```clojure
(let [request (build-request httpRequest)
      ...
```